htdocs/core/class/html.form.class.php

```diff
@@ -2063,6 +2064,8 @@ function select_produits_list($selected='',$htmlname='productid',$filtertype='',
 		if (!empty($conf->global->PRODUIT_ATTRIBUTES_HIDECHILD)) {
 			$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."product_attribute_combination pac ON pac.fk_product_child = p.rowid";
 		}
+		//Product category
+		$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."categorie_product as pcat ON pcat.fk_product=p.rowid";
```

Comment:
If a product has several different categories (most common case when tags are used), this will duplicate record of products. Also what should be the position if product has both tag AAA and tag ZZZ

Reply:
I propose to eliminate duplicates in this way:

```
SELECT p.rowid, p.label, p.ref, p.description, p.barcode, p.fk_product_type, p.price, p.price_ttc, p.price_base_type, p.tva_tx, p.duration, p.fk_price_expression, p.stock
	,(
		SELECT xxx_categorie_product.fk_categorie
		FROM xxx_categorie_product
		WHERE xxx_categorie_product.fk_product=p.rowid 
		LIMIT 1
	) AS categorie_product_id

FROM xxx_product as p
WHERE p.entity IN (1) AND p.tosell = 1 
ORDER BY categorie_product_id DESC 
LIMIT 1000		
```

I didn't find a rowid in the xxx_categorie_product table to check (by default or by a parameter/const) the first category assigned to the product.

what do you think?